compliance/cis_k8s/rules/cis_4_2_11/data.yaml

```diff
@@ -0,0 +1,78 @@
+metadata:
+  id: 5fea55c8-4d83-55a4-afc5-9aa850f39ab9
+  name: Ensure that the --rotate-certificates argument is not set to false(Manual)
+  profile_applicability: |
+    • Level 1 - Worker Node
+  description: |
+    Enable kubelet client certificate rotation.
+  rationale: >
+    The `--rotate-certificates` setting causes the kubelet to rotate its client
+    certificates by creating new CSRs as its existing credentials expire. 
+    This automated periodic rotation
+
+    ensures that the there is no downtime due to expired certificates and thus addressing
+
+    availability in the CIA security triad.
+
+    Note: This recommendation only applies if you let kubelets get their certificates from the
+
+    API server. In case your kubelet certificates come from an outside authority/tool (e.g.
+
+    Vault) then you need to take care of rotation yourself.
+
+    Note: This feature also require the `RotateKubeletClientCertificate` feature gate to be
+
+    enabled (which is the default since Kubernetes v1.7)
+  audit: |
+    Run the following command on each node:
+    ```
+    ps -ef | grep kubelet
+    ```
+    
+    Verify that the `--rotate-certificates` argument is not present, or is set to `true`.
+
+    If the `--rotate-certificates` argument is not present, verify that if there is a Kubelet
+
+    config file specified by `--config`, that file does not contain `rotateCertificates: false`.
+  remediation: |
+    If using a Kubelet config file, edit the file to add the line
+    `rotateCertificates: true` or
+
+    remove it altogether to use the default value.
+
+    If using command line arguments, edit the kubelet service file
+
+    `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf` on each worker node and
+
+    remove `--rotate-certificates=false` argument from the `KUBELET_CERTIFICATE_ARGS`
+
+    variable.
+
+    Based on your system, restart the kubelet service. For example:
+    ```
+    systemctl daemon-reload
+
+    systemctl restart kubelet.service
+    ```
+  impact: |
+    None
+  default_value: |
+    By default, kubelet client certificate rotation is enabled.
+  references: |
+    1. https://github.com/kubernetes/kubernetes/pull/41912
```

Comment:
```suggestion
    1. [https://github.com/kubernetes/kubernetes/pull/41912](https://github.com/kubernetes/kubernetes/pull/41912)
```

Reply:
This needs to be done for all links

Reply:
@jeniawhite isn't it redundant for urls?

Reply:
I've followed the following guide:
https://elastic.github.io/eui/#/editors-syntax/markdown-format

Notice how they are building the links:
<img width="909" alt="Screen Shot 2022-04-27 at 15 10 12" src="https://user-images.githubusercontent.com/12246907/165515181-686ad80b-e671-4256-9104-c4a7cb9d704e.png">

Reply:
@jeniawhite the example also excluded URLs from what I see? i.e., there's no need to explicitly type URL as a linkable part, it will happen automatically, you can check it in any md reader online, see:
```
www.my-url.com
```
www.my-url.com
```
[www.my-url.com](www.my-url.com)
```
[www.my-url.com](www.my-url.com)